docs/guides/4-adding-redirect-rules-to-routes.md

```diff
@@ -0,0 +1,104 @@
+# Redirect Rules
+
+The Routes configuration object allows a user to specify a redirection from a route. It also allows redirection to be conditional.
+
+Redirection at the route config level is super easy with Yarr. The code sandbox below covers various redirection examples mentioned in this guide. ⬇️
+
+[Open Codesandbox](https://codesandbox.io/embed/typescript-react-yarr-redirectrules-example-m1nyu?fontsize=14&hidenavigation=1&theme=dark)
+
+## Always redirect from one path to another path
+
+There are scenarios where we want to avoid a user from ever landing on a path. It could be for the following reasons
+
+- Feature at the page is deprecated
+- Page now has a new location
```

Comment:
Another use case is authentication.

You could deterministically redirect users from one route to another if they were/weren't authenticated.

Reply:
@eric

Reply:
@erictaylor I added an authentication example below already.